panos_upgrade_assurance/firewall_proxy.py

```diff
@@ -871,23 +871,28 @@ def get_disk_utilization(self) -> dict:
             row_items_trimmed = [item for item in row_items if item != ""]
 
             mount_point = row_items_trimmed[-1]
-            free_size_short = row_items_trimmed[3]
-            free_size_name = free_size_short[-1]
-            free_size_number = float(free_size_short[0:-1])
-
-            if isinstance(free_size_name, str):
-                if free_size_name == "G":
-                    free_size = free_size_number * 1024
-                elif free_size_name == "M":
-                    free_size = free_size_number
-                elif free_size_name == "K":
-                    free_size = free_size_number / 1024
-
-            elif isinstance(free_size_name, int):
-                free_size = free_size_short / 1024 / 1024
-
-            else:
-                raise exceptions.WrongDiskSizeFormatException("Free disk size has wrong format.")
+            try:
+                free_size_short = row_items_trimmed[3]
+                free_size_name = free_size_short[-1]
+                free_size_number = float(free_size_short[0:-1])
+
+                if not free_size_name.isnumeric():
+                    if free_size_name == "G":
+                        free_size = free_size_number * 1024
+                    elif free_size_name == "M":
+                        free_size = free_size_number
+                    elif free_size_name == "K":
+                        free_size = free_size_number / 1024
+                    else:
+                        raise Exception("Free disk size has wrong format.")
+                else:
+                    free_size = float(free_size_short) / 1024 / 1024
+
+            except Exception as exp:
+                if str(exp) == "Free disk size has wrong format.":
+                    raise exceptions.WrongDiskSizeFormatException(str(exp))
+                else:
+                    raise exceptions.WrongDiskSizeFormatException("API response has wrong format.")
```

Comment:
I would suggest we raise different exceptions here for API response and wrong format. For API response being in wrong format we can raise `MalformedResponseException` which we have in several places and for the wrong disk size format exception we raise it as is.

```suggestion
                        raise exceptions.WrongDiskSizeFormatException("Free disk size has wrong format.")
                else:
                    free_size = float(free_size_short) / 1024 / 1024
            except exceptions.WrongDiskSizeFormatException:           
                raise
            except Exception:
                raise exceptions.MalformedResponseException(
                    f"Reported disk space block does not have typical structure: <{row}>."
                )
```
**Note: I haven't tested the suggested code.**

Reply:
👍